Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -90,13 +90,17 @@ \subsection{Moodcat Backend}
 \begin{itemize}
 \item the \textit{Static File Server} that serves the frontend website on a webserver;
 \item the \textit{REST API} which is used for communication about rooms and songs between the backend and frontend;
-\item the \textit{music matching algorithm} which is used to generate music suggestions for rooms and to find a room for a user.
+\item the \textit{Music Matching Algorithm} which is used to generate music suggestions for rooms and to find a room for a user.
 \end{itemize}
 In the following paragraphs we will explain the last two subcomponents.
 
 \subsubsection{REST API}
 The backend keeps track of which rooms are active and what song is currently played in the room.
-The API can process user votes and classifications in order to improve the system.
+When the backed is started, the backend retrieves rooms from the database and creates an instances for each of those.
+The backend keeps track of a current song, a history of previously played songs and a playing queue for future tracks to play.
```

Comment:
"For each room instance, the backend keeps track of the current song, the history" (replace a with the and state that the room instances store them)